scripts/release/asset_publish.py

```diff
@@ -277,33 +289,65 @@ def _str2bool(v: str) -> bool:
             )
 
             # Handle specific asset types
-            if asset.type in [assets.AssetType.COMPONENT, assets.AssetType.ENVIRONMENT]:
-                # Assemble command
-                cmd = assemble_command(
-                    asset.type.value, str(asset.spec_with_path),
-                    registry_name, final_version, resource_group, workspace, debug_mode)
-                # Run command
-                run_command(cmd, failure_list, debug_mode)
+            if asset.type == assets.AssetType.COMPONENT:
+                # load component and check if environment exists
+                logger.print(f"spec's path: {asset.spec_with_path}")
+                component = load_component(asset.spec_with_path)
+                env = component.environment
+
+                match = REGISTRY_ENV_PATTERN.match(env)
+                if not match:
+                    match = WORKSPACE_ENV_PATTERN.match(env)
+                    if not match:
+                        logger.print(
+                            "Env ID doesn't match workspace or registry pattern"
+                            + f" in {asset.spec_with_path}"
+                        )
+                        failure_list.append(asset)
+                        continue
+
+                # both ws and registry env follows the same grouping
+                env_name = match.group(1)
+                env_version = match.group(2)
+                logger.print(f"Env name:version => {env_name}:{env_version}")
+                logger.print(f"Using final_version {final_version} to check if env registered")
+                try:
+                    mlclient = MLClient(
+                        DefaultAzureCredential(),
+                        subscription_id=subscription_id,
+                        registry_name=registry_name,
+                    )
+                    env = mlclient.environments.get(name=env_name, version=final_version)
```

Comment:
This approach will work for environments in azureml-preview and azureml (prod), where there aren't any timestamp/release-based suffixes added to versions. Note that for now environments in azureml need to be released via a separate pipeline first, then the component that references it can be released.

Reply:
thank you Louie. I have updated code path to check for env version mentioned in the spec file and final version calculated as part of asset publish process. It should cover both the use cases. Can you please check once more and help to review?Thanks